src/Microsoft.Net.Http.Headers/HeaderUtilities.cs

```diff
@@ -208,6 +295,82 @@ public static bool TryParseInt64(string value, out long result)
             return long.TryParse(value, NumberStyles.None, NumberFormatInfo.InvariantInfo, out result);
         }
 
+        internal static unsafe bool TryParseInt32(StringSegment value, out int result)
+        {
+            if (string.IsNullOrEmpty(value.Buffer))
+            {
+                result = 0;
+                return false;
+            }
+
+            result = 0;
+            fixed (char* ptr = value.Buffer)
+            {
+                var ch = (ushort*)ptr;
+                ch += value.Offset;
+                var end = ch + value.Length;
+
+                if (ch == end)
+                {
+                    result = 0;
+                    return false;
+                }
+
+                ushort digit = 0;
+                while (ch < end && (digit = (ushort)(*ch - 0x30)) <= 9)
+                {
+                    result *= 10;
+                    result += digit;
+                    ch++;
+                }
+
+                if (ch != end)
+                {
+                    result = 0;
+                    return true;
+                }
+                return true;
+            }
+        }
+
+        public static unsafe bool TryParseInt64(StringSegment value, out long result)
```

Comment:
I was testing it through TryParseTimeSpan but I'll add some tests specific to this method.